src/module/macros/attack.ts

```diff
@@ -0,0 +1,593 @@
+// Import TypeScript modules
+import { LANCER } from "../config";
+import { getAutomationOptions } from "../settings";
+import type { LancerItem } from "../item/lancer-item";
+import type { LancerActor } from "../actor/lancer-actor";
+import { is_reg_mech } from "../actor/lancer-actor";
+import type {
+  LancerAttackMacroData,
+  LancerMacroData,
+} from "../interfaces";
+import {
+  DamageType,
+  funcs,
+  MechWeapon,
+  MechWeaponProfile,
+  NpcFeature,
+  OpCtx,
+  Pilot,
+  PilotWeapon,
+  RegRef,
+} from "machine-mind";
+import { is_limited, is_overkill } from "machine-mind/dist/funcs";
+import { is_loading, is_self_heat } from "machine-mind/dist/classes/mech/EquipUtil";
+import { FoundryReg } from "../mm-util/foundry-reg";
+import { checkForHit } from "../helpers/automation/targeting";
+import type { AccDiffData, AccDiffDataSerialized, RollModifier } from "../helpers/acc_diff";
+import { getMacroSpeaker, encodeMacroData, ownedItemFromString } from "./util"
+import { renderMacroTemplate } from "./render"
+import { rollReactionMacro } from "./reaction"
+import { rollSystemMacro } from "./system"
+import { rollTalentMacro } from "./talent"
+import { prepareTechMacro } from "./tech"
+import { rollTextMacro } from "./text"
+import { rollTriggerMacro } from "./trigger"
+
+const lp = LANCER.log_prefix;
+
+function rollStr(bonus: number, total: number): string {
+  let modStr = "";
+  if (total != 0) {
+    let sign = total > 0 ? "+" : "-";
+    let abs = Math.abs(total);
+    let roll = abs == 1 ? "1d6" : `${abs}d6kh1`;
+    modStr = ` ${sign} ${roll}`;
+  }
+  return `1d20 + ${bonus}${modStr}`;
+}
+
+function applyPluginsToRoll(str: string, plugins: RollModifier[]): string {
+  return plugins.sort((p, q) => q.rollPrecedence - p.rollPrecedence).reduce((acc, p) => p.modifyRoll(acc), str);
+}
+
+type AttackRolls = {
+  roll: string;
+  targeted: {
+    target: Token;
+    roll: string;
+    usedLockOn: { delete: () => void } | null;
+  }[];
+};
+
+export function attackRolls(bonus: number, accdiff: AccDiffData): AttackRolls {
+  let perRoll = Object.values(accdiff.weapon.plugins);
+  let base = perRoll.concat(Object.values(accdiff.base.plugins));
+  return {
+    roll: applyPluginsToRoll(rollStr(bonus, accdiff.base.total), base),
+    targeted: accdiff.targets.map(tad => {
+      let perTarget = perRoll.concat(Object.values(tad.plugins));
+      return {
+        target: tad.target,
+        roll: applyPluginsToRoll(rollStr(bonus, tad.total), perTarget),
+        usedLockOn: tad.usingLockOn,
+      };
+    }),
+  };
+}
+
+
+type AttackMacroOptions = {
+  accBonus: number;
+  damBonus: { type: DamageType; val: number };
```

Comment:
Note to self - where do we use this? Shouldn't `damBonus.val` be a string, since it's usually "1d6"?

Reply:
I'm pretty sure this is for flat damage bonuses, not the actual damage roll. That said, I had a quick look and it seems we only have code for consuming this option, and nothing that generates it.